lib/Paws/Net/Caller.pm

```diff
@@ -29,7 +29,7 @@ package Paws::Net::Caller;
         (defined $requestObj->content)?(content => $requestObj->content):(),
       }
     );
-    return ($response->{status}, $response->{content}, $response->{headers});
+    return $self->caller_to_response($service, $call_object, $response->{status}, $response->{content}, $response->{headers});
```

Comment:
Sorry for that. Just that on the s3_support branch I merged in the "latest and greatest" state I could, and that includes a small refactor so that we can have a "MockCaller" that records calls to the APIs, and remembers the input and outputs so they can later be replayed. It's used here, and maybe you will find it useful: https://github.com/pplu/aws-sdk-perl/blob/test_recorder_refactor/t/11_client_exceptions.t